tower_cli/models/base.py

```diff
@@ -1208,3 +1206,45 @@ def modify(self, pk=None, create_on_missing=False, **kwargs):
         """
         return self.write(pk, create_on_missing=create_on_missing,
                           force_on_exists=True, **kwargs)
+
+
+class SurveyResource(Resource):
+    """Contains utilities and commands common to "job template" models,
+    which take extra_vars and have a survey_spec."""
+    abstract = True
+
+    def _survey_endpoint(self, pk):
+        return '{0}{1}/survey_spec/'.format(self.endpoint, pk)
+
+    def write(self, pk=None, **kwargs):
+        survey_input = kwargs.pop('survey_spec', None)
+        if kwargs.get('extra_vars', None):
+            kwargs['extra_vars'] = parser.process_extra_vars(
+                kwargs['extra_vars'])
+        ret = super(SurveyResource, self).write(pk=pk, **kwargs)
+        if survey_input is not None and ret.get('id', None):
+            if not isinstance(survey_input, dict):
+                survey_input = json.loads(survey_input.strip(' '))
+            if survey_input == {}:
+                debug.log('Deleting the survey_spec.', header='details')
+                r = client.delete(self._survey_endpoint(ret['id']))
+            else:
+                debug.log('Saving the survey_spec.', header='details')
+                r = client.post(self._survey_endpoint(ret['id']),
+                                data=survey_input)
+            if r.status_code == 200:
+                ret['changed'] = True
+            if survey_input and not ret['survey_enabled']:
+                debug.log('For survey to take effect, set survey_enabled'
+                          ' field to True.', header='warning')
+        return ret
+
+    @resources.command
```

Comment:
One of the values of model/command-line separation comes here: `pk` is the only field needed for `survey` command, but all model fields automatically fill `kwargs`, which makes code confusing and pollute help text.